services/slam/builtin/builtin_test.go

```diff
@@ -548,17 +550,20 @@ func TestGeneralNew(t *testing.T) {
 	})
 
 	t.Run("New slam service with no camera", func(t *testing.T) {
+		port1, err := utils.TryReserveRandomPort()
+		test.That(t, err, test.ShouldBeNil)
 		attrCfg := &builtin.AttrConfig{
 			Sensors:       []string{},
 			ConfigParams:  map[string]string{"mode": "2d"},
 			DataDirectory: name,
-			Port:          "localhost:4445",
+			Port:          "localhost:" + strconv.Itoa(port1),
```

Comment:
if this is migrated into a go module that still execs C++, we should really have the C++ start up and tell us the port instead of reserving since there's always a port reserve race.

Reply:
The SLAM service is set up in a way where if the Port attribute is not specified, we parse the logs of the C++ server until we find which port it reserves. If the port is specified, we pass through the value.

Once we start building unit tests for the modules I 100% agree that the reserved port should happen within the C++ server. Currently, we only do this for integration tests

Reply:
Even when this test is migrated into the go module, it will still only run a test gRPC server and not the actual C++ binary. John is correct that for the integration tests, where we do run the C++ binary, the C++ binary reports the port

Reply:
why not just remove the port then?

Reply:
setupTestGRPCServer requires a port

Reply:
o cool, I'll def swap to using that. Should I just return `listener2.Addr().String()` so that the slam service client knows where to look? or is there a better way to grab the port from here?

Reply:
yeah the setup can return an extra value of int from `listener.Addr().(*net.TCPAddr).Port`. so `return gServer2, listener.Addr().(*net.TCPAddr).Port`

I'd suggest making the code this actually:
```func setupTestGRPCServer(tb testing.TB) (*grpc.Server, int) {
	listener, err := net.Listen("tcp", ":0")
	test.That(tb, err, test.ShouldBeNil)
	grpcServer := grpc.NewServer()
	go grpcServer.Serve(listener)

	return grpcServer, listener.Addr().(*net.TCPAddr).Port
}```

Reply:
kk updated code to reflect this